BDArmory/Ammo/PooledBullet.cs

```diff
@@ -869,18 +913,22 @@ bool BulletHitAnalysis(RaycastHit hit, float period)
             KerbalEVA hitEVA;
             try
             {
-                hitPart = hit.collider.gameObject.GetComponentInParent<Part>();
-                hitEVA = hit.collider.gameObject.GetComponentUpwards<KerbalEVA>();
+                hitPart = bulletHit.hit.collider.gameObject.GetComponentInParent<Part>();
+                hitEVA = bulletHit.hit.collider.gameObject.GetComponentUpwards<KerbalEVA>();
             }
             catch (NullReferenceException e)
             {
                 Debug.Log("[BDArmory.PooledBullet]:NullReferenceException for Ballistic Hit: " + e.Message);
                 return true;
             }
 
-            if (hitPart != null && ProjectileUtils.IsIgnoredPart(hitPart)) return false; // Ignore ignored parts.
-            if (hitPart != null && hitPart == sourceWeapon) return false; // Ignore weapon that fired the bullet.
-            if (hitPart != null && (hitPart == CurrentPart && ProjectileUtils.IsArmorPart(CurrentPart))) return false; //only have bullet hit armor panels once - no back armor to hit if penetration
+            if (hitPart != null)
+            {
+                if (ProjectileUtils.IsIgnoredPart(hitPart)) return false; // Ignore ignored parts.
+                if (hitPart == sourceWeapon) return false; // Ignore weapon that fired the bullet.
+                if (bulletHit.isReverseHit && (hitPart == CurrentPart && ProjectileUtils.IsArmorPart(CurrentPart))) return false; //only have bullet hit armor panels once - no back armor to hit if penetration
```

Comment:
I think we can drop the `hitPart == CurrentPart && ` check here now since the assumption is that if it's a reverse hit, then the forward hit was already accounted for on this frame or the previous one and the `hitPart == CurrentPart` condition was what was causing the problem for adjacent armour plates in the first place.